### PR TITLE
Refactor and add functional components

### PR DESCRIPTION
**Pull request by steinalex**
All components have been made into functional components, as well as some slight refactoring of the search component

### CODE REVIEW

File: src/features/company/components/Company.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const Company = () => {
         loaded={selectedCompanyOverview !== undefined}
         render={renderCompanyComponent}
       />
-      <Peers peers={selectedTopPeers} />
+      <Peers peers={selectedTopPeers && selectedTopPeers} />
```
Comment (BenGladman) [inline src/features/company/components/Company.tsx]: The added `&& selectedTopPeers` makes no difference here

File: src/features/market-status/components/MarketStatus.tsx [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
-export const MarketStatus = ({ stock, keyStats }: any) => {
+type MarketStatusProps = {
+  stock: IStockTicker | undefined;
+  keyStats: IKeyStats | null;
+};
+
+export const MarketStatus: FC<MarketStatusProps> = ({
+  stock,
+  keyStats
+}: any) => {
```
Comment (BenGladman) [inline src/features/market-status/components/MarketStatus.tsx]: You can remove the "any"

File: src/features/company/components/Company.tsx (hunk 1 above)
Comment (steinalex) [reply src/features/company/components/Company.tsx]: Good spot, will remove in a new PR

File: src/features/market-status/components/MarketStatus.tsx (hunk 2 above)
Comment (steinalex) [reply src/features/market-status/components/MarketStatus.tsx]: Good spot, will remove in a new PR

**Pull request merged**